File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/context/fight/GameFightFighterInformations.scala





















// Generated on 05/03/2013 10:35:04
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait GameFightFighterInformations extends GameContextActorInformations {
  override val typeId = 143

  protected trait Type extends GameContextActorInformations.Type with Serializable {
    override val typeId = GameFightFighterInformations.typeId

    val teamId: Byte
    val alive: Boolean
    val stats: GameFightMinimalStats.TypeImpl

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeByte(teamId)
      buf.writeBoolean(alive)
      buf.writeUShort(stats.typeId)
      stats.serialize(buf)
    }

  }

  protected def create(contextualId: Int, look: EntityLook.TypeImpl, disposition: EntityDispositionInformations.TypeImpl, teamId: Byte, alive: Boolean, stats: GameFightMinimalStats.TypeImpl): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val contextualId = buf.readInt()
    val look = EntityLook.deserialize(buf)
    val disposition = types.get[EntityDispositionInformations.TypeImpl](buf.readShort());
    val teamId = buf.readByte()
    val alive = buf.readBoolean()
    val stats = types.get[GameFightMinimalStats.TypeImpl](buf.readShort());

    create(contextualId, look, disposition, teamId, alive, stats)
  }

}

object GameFightFighterInformations extends GameFightFighterInformations {
  case class TypeImpl(contextualId: Int, look: EntityLook.TypeImpl, disposition: EntityDispositionInformations.TypeImpl, teamId: Byte, alive: Boolean, stats: GameFightMinimalStats.TypeImpl) extends Type
  override def create(contextualId: Int, look: EntityLook.TypeImpl, disposition: EntityDispositionInformations.TypeImpl, teamId: Byte, alive: Boolean, stats: GameFightMinimalStats.TypeImpl) = TypeImpl(contextualId, look, disposition, teamId, alive, stats)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(contextualId: Int, look: EntityLook.TypeImpl, disposition: EntityDispositionInformations.TypeImpl, teamId: Byte, alive: Boolean, stats: GameFightMinimalStats.TypeImpl) = create(contextualId, look, disposition, teamId, alive, stats)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/GameFightFighterInformations.java


// Generated on 05/08/2013 19:38:00
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class GameFightFighterInformations extends GameContextActorInformations {
    public static final short TYPE_ID = 143;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public byte teamId;
    public boolean alive;
    public GameFightMinimalStats stats;
    
    public GameFightFighterInformations() { }
    
    public GameFightFighterInformations(int contextualId, EntityLook look, EntityDispositionInformations disposition, byte teamId, boolean alive, GameFightMinimalStats stats) {
        super(contextualId, look, disposition);
        this.teamId = teamId;
        this.alive = alive;
        this.stats = stats;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeByte(teamId);
        buf.writeBoolean(alive);
        buf.writeShort(stats.getTypeId());
        stats.serialize(buf);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        teamId = buf.readByte();
        if (teamId < 0)
            throw new RuntimeException("Forbidden value on teamId = " + teamId + ", it doesn't respect the following condition : teamId < 0");
        alive = buf.readBoolean();
        stats = ProtocolTypeManager.getInstance().build(buf.readShort());
        stats.deserialize(buf);
    }
    
}
